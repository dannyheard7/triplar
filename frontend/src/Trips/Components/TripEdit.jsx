import React from "react";

import TripForm from "./TripForm";
import { connect } from "react-redux";
import {editTrip} from "../utils/actions";
import ReduxFormContainer from "../../Forms/Containers/ReduxFormContainer";

export class TripEdit extends React.Component {
    curryAction = (action, id) => (object) => action(id, object);

    render() {
        const errors = this.props.errors ? this.props.errors : [];
        let action = this.curryAction(editTrip, this.props.trip.id);

        return (
            <div className="card">
                <ReduxFormContainer action={action} >
                    <TripForm submitLabel="Save" trip={this.props.trip} fieldErrors={errors}/>
                </ReduxFormContainer>
            </div>
        );
    }
}

TripEdit.defaultProps = {
    trip: {}
};

function mapStateToProps(state, ownProps) {
  return {
      trip: state.trips.trips.find(x => x.id === ownProps.match.params.id),
      errors: state.errors.EDIT_TRIP
  };
}

export default connect(mapStateToProps)(TripEdit)