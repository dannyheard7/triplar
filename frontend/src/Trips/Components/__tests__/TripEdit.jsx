import React from "react";
import {shallow} from "enzyme";

import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();

import {TripEdit} from "../TripEdit";

jest.mock('../../utils/trips.api.js');
const api = require('../../utils/trips.api.js');
const faker = require('faker');

describe('<TripEditContainer />', () => {
    test('renders correctly', () => {
        const result = renderer.render(<TripEdit />);
        expect(result).toMatchSnapshot();
    });

    test.skip('editTrip called with correct arguments', () => {
        const spy = jest.spyOn(api.default, 'editTrip');
        const trip = {id: faker.random.number(), name: faker.random.word()};

        const container = shallow(<TripEdit trip={trip} />);
        container.instance().apiFunc(trip);

        expect(spy).toBeCalledWith(trip.id, trip);

        spy.mockRestore();
    });
});