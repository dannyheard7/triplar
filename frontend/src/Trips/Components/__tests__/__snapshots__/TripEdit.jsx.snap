// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripEditContainer /> renders correctly 1`] = `
<div
  className="card"
>
  <Connect(ReduxFormContainer)
    action={[Function]}
  >
    <TripForm
      fieldErrors={Array []}
      submitLabel="Save"
      trip={Object {}}
    />
  </Connect(ReduxFormContainer)>
</div>
`;
